keiko/keiko/keiko.py

```diff
@@ -135,32 +168,41 @@ def generate_report(
 
         # run pdflatex
         cmd = [
-            "pdflatex",
+            "latexmk",
+            "-xelatex",
             "-synctex=1",
             "-interaction=nonstopmode",
             preprocessed_tex_path.as_posix(),
         ]
         env = {**os.environ, "TEXMFVAR": directory}
-        for i in (1, 2):
-            output = subprocess.run(cmd, cwd=directory, env=env, capture_output=True, check=False)
-            current_span.add_event(f"Completed pdflatex run {i}")
+
+        def log_output(level, output):
+            if not logger.isEnabledFor(level):
+                return
+            # prefix all lines in output
+            for line in output.decode("utf-8").splitlines():
+                logger.log(level, "latexmk output: %s", line)
```

Comment:
Can we also log the pid or some other thing unique to this output? With these kind of multiline logs I think there is no guarantee that output from multiple requests aren't logged at the same time which might get confusing.

Reply:
Good suggestion, I've added the `report_id` which should be unique.